### PR TITLE
Introduce EmptyStateComponent

### PR DESCRIPTION
**Pull request by jespr**
This implements an EmptyStateComponent for #201 - a bit unsure how to best test the inline svg

### Pull request checklist

- [x] Your code contains tests relevant for code you modified
- [x] You have linted and tested the project with `bin/check`

### CODE REVIEW

File: app/components/empty_state_component.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
```
Comment (joemasilotti) [inline app/components/empty_state_component.rb:1]: Please remove this comment.

File: app/views/notifications/index.html.erb [hunk 2]
```diff
@@ -4,7 +4,7 @@
     <%= render @notifications %>
   <% else %>
     <div class="mt-8">
-      <%= render "empty_state" %>
+      <%= render EmptyStateComponent.new(title: t("notifications.empty_state.title"), body: t("notifications.empty_state.body"), icon: "icons/outline/bell.svg") %>
```
Comment (joemasilotti) [inline app/views/notifications/index.html.erb:7]: Let's move the translations so they aren't so long to access. For example,

```yaml
en:
  notifications:
    index:
      empty_state:
        title:
        body:
```

Then we can use `t(".empty_state.title")` and `t(".empty_state.body")` here.

File: test/components/empty_state_component_test.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+class EmptyStateComponentTest < ViewComponent::TestCase
+  test "Renders the title, body and icon" do
+    title = "No notifications"
+    body = "Nothing to see"
+
+    render_inline(EmptyStateComponent.new(title: title, body: body, icon: "icons/outline/bell.svg"))
+
+    assert_selector("h3", text: title)
+    assert_selector("p", text: body)
+    assert_selector("svg")
```
Comment (joemasilotti) [inline test/components/empty_state_component_test.rb:12]: If you use an SVG from `icons/brands/` they have the `title` attribute set. And you can test them like `test/components/social_link_component_test.rb`.

**Review by joemasilotti — CHANGES_REQUESTED** [review]
Thanks for the PR! I requested a few changes and commented on how you can test the SVG.

File: app/views/notifications/index.html.erb (hunk 2 above)
Comment (jespr) [reply app/views/notifications/index.html.erb:7]: I went with this approach first, but wasn't sure if there was previous history with how it was placed in `en.yml` - will change this!

File: test/components/empty_state_component_test.rb (hunk 3 above)
Comment (jespr) [reply test/components/empty_state_component_test.rb:12]: Oh!! Neat

**Comment by jespr** [comment]
@joemasilotti should be good now :)

**Review by joemasilotti — APPROVED** [review]

**Comment by joemasilotti** [comment]
Awesome! Thanks for the contribution.

**Pull request merged**